codegen/core/src/main/java/software/amazon/smithy/python/codegen/generators/ConfigGenerator.java

```diff
@@ -138,15 +138,17 @@ private static List<ConfigProperty> getHttpProperties(GenerationContext context)
         if (usesHttp2(context)) {
             clientBuilder
                     .initialize(writer -> {
-                        writer.addDependency(SmithyPythonDependency.SMITHY_HTTP.withOptionalDependencies("awscrt"));
+                        writer.addDependency(SmithyPythonDependency.SMITHY_HTTP);
+                        writer.addDependency(SmithyPythonDependency.AWS_CRT);
                         writer.addImport("smithy_http.aio.crt", "AWSCRTHTTPClient");
                         writer.write("self.http_client = http_client or AWSCRTHTTPClient()");
                     });
 
         } else {
             clientBuilder
                     .initialize(writer -> {
-                        writer.addDependency(SmithyPythonDependency.SMITHY_HTTP.withOptionalDependencies("aiohttp"));
+                        writer.addDependency(SmithyPythonDependency.SMITHY_HTTP);
+                        writer.addDependency(SmithyPythonDependency.AIO_HTTP);
```

Comment:
rereading this, I think this is slightly incorrect, but assuming the other conditional triggers now isn't a hard blocker for right now.

What we want this to look like eventually is:

`smithy-http[awscrt]`
or
`smithy-http[aiohttp]`

where the http client version is declared in smithy-http and we're not taking additional dependencies. I think we can track that as follow up wort, I'm not sure `.withOptionalDependencies("aiohttp")` was actually working before.